src/biome/text/dataset.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import multiprocessing
+import pickle
+from typing import Union, Dict, Iterable, List
+
+import datasets
+from allennlp.data import AllennlpDataset, AllennlpLazyDataset, Instance
+
+InstancesDataset = Union[AllennlpDataset, AllennlpLazyDataset]
+
+
+class Dataset:
+    """A dataset to be used with biome.text Pipelines
+
+    Basically a light wrapper around HuggingFace's `datasets.Dataset`
+
+    Parameters
+    ----------
+    dataset
+        A HuggingFace `datasets.Dataset`
+    """
+
+    _LOGGER = logging.getLogger(__name__)
+    _PICKLED_INSTANCES_COL_NAME = "PICKLED_INSTANCES_FOR_BIOME_PIPELINE"
+
+    def __init__(
+        self, dataset: datasets.Dataset,
+    ):
+        self.dataset: datasets.Dataset = dataset
+
+    @classmethod
+    def load_dataset(cls, *args, split, **kwargs) -> "Dataset":
+        """Load a dataset using Huggingface's `datasets.load_dataset` method.
+
+        See https://huggingface.co/docs/datasets/loading_datasets.html
+
+        Parameters
+        ----------
+        split
+            See https://huggingface.co/docs/datasets/splits.html
+        *args/**kwargs
+            Passed on to the `dataset.load_dataset` method
+
+        Returns
+        -------
+        dataset
+        """
+        return cls(datasets.load_dataset(*args, split=split, **kwargs))
+
+    @classmethod
+    def from_json(cls, paths: Union[str, List[str]], **kwargs) -> "Dataset":
+        """Convenient method to create a Dataset from a json file
+
+        Parameters
+        ----------
+        paths
+            One or several paths to json files
+        **kwargs
+            Passed on to the `datasets.load_dataset` method
+
+        Returns
+        -------
+        dataset
+        """
+        return cls.load_dataset("json", data_files=paths, split="train", **kwargs)
+
+    @classmethod
+    def from_csv(cls, paths: Union[str, List[str]], **kwargs) -> "Dataset":
+        """Convenient method to create a Dataset from a csv file
+
+        Parameters
+        ----------
+        paths
+            One or several paths to csv files
+        **kwargs
+            Passed on to the `datasets.load_dataset` method
+
+        Returns
+        -------
+        dataset
+        """
+        return cls.load_dataset("csv", data_files=paths, split="train", **kwargs)
+
+    def to_instances(self, pipeline: "biome.text.Pipeline", lazy=True) -> InstancesDataset:
+        """Convert input to instances for the pipeline
+
+        Parameters
+        ----------
+        pipeline
+            The pipeline for which to create the instances.
+        lazy
+            If true, instanes are lazily read from disk, otherwise they are kept in memory.
+        """
+        self._LOGGER.info("Creating instances ...")
+
+        input_columns = [col for col in pipeline.inputs + [pipeline.output] if col]
```

Comment:
using new datasets, you must adapt dataset format to pipeline inputs/outputs isn't it?

Reply:
If you mean the mapping, yes, one has to make sure that the dataset contains the "right" column names. But this can be easily done with the Huggingface `Dataset` methods, for example:
```python
ds.dataset.rename_column_("old_col_name", "new_col_name")
# or combining stuff
ds.datasest = ds.dataset.map(lambda x: {"new_col_name": x["old_col_name"] + x["old_col_name2"]})
```

Reply:
cool !